src/components/views/settings/AddPhoneNumber.js

```diff
@@ -119,52 +125,47 @@ class AddPhoneNumber extends React.Component {
                 }).catch((err) => {
                     this._promptForMsisdnVerificationCode(msisdn, err);
                 }).finally(() => {
+                    if (this._unmounted) return;
                     this.setState({msisdn_add_pending: false});
                 }).done();
             }
         });
-    }
+    },
 
-    render() {
+    render: function() {
         const Loader = sdk.getComponent("elements.Spinner");
         if (this.state.msisdn_add_pending) {
             return <Loader />;
-        } else if (!this.props.matrixClient.isGuest()) {
-            const CountryDropdown = sdk.getComponent('views.login.CountryDropdown');
-            // XXX: This CSS relies on the CSS surrounding it in UserSettings as its in
-            // a tabular format to align the submit buttons
-            return (
-                <form className="mx_UserSettings_profileTableRow" onSubmit={this._onAddMsisdnSubmit}>
-                    <div className="mx_UserSettings_profileLabelCell">
-                    </div>
-                    <div className="mx_UserSettings_profileInputCell">
-                        <div className="mx_Login_phoneSection">
-                            <CountryDropdown onOptionChange={this._onPhoneCountryChange}
-                                className="mx_Login_phoneCountry"
-                                value={this.state.phoneCountry}
-                            />
-                            <input type="text"
-                                ref={this._collectAddMsisdnInput}
-                                className="mx_UserSettings_phoneNumberField"
-                                placeholder="Add phone number"
-                                value={this.state.phoneNumber}
-                                onChange={this._onPhoneNumberChange}
-                            />
-                        </div>
-                    </div>
-                    <div className="mx_UserSettings_threepidButton mx_filterFlipColor">
-                         <input type="image" value="Add" src="img/plus.svg" width="14" height="14" />
-                    </div>
-                </form>
-            );
+        } else if (this.props.matrixClient.isGuest()) {
+            return null;
```

Comment:
I don't think you're allowed to return `null` from a `render`. Return an empty div or sth instead.